strawberry/schema/types/scalar.py

```diff
@@ -42,17 +48,23 @@ def _make_scalar_definition(scalar_type: GraphQLScalarType) -> ScalarDefinition:
 
 
 DEFAULT_SCALAR_REGISTRY: Dict[object, ScalarDefinition] = {
-    type(None): base_scalars.Void._scalar_definition,
-    None: base_scalars.Void._scalar_definition,
+    type(None): cast(_HasScalarDefinition, base_scalars.Void)._scalar_definition,
+    None: cast(_HasScalarDefinition, base_scalars.Void)._scalar_definition,
     str: _make_scalar_definition(GraphQLString),
     int: _make_scalar_definition(GraphQLInt),
     float: _make_scalar_definition(GraphQLFloat),
     bool: _make_scalar_definition(GraphQLBoolean),
     ID: _make_scalar_definition(GraphQLID),
-    UUID: base_scalars.UUID._scalar_definition,
-    Upload: Upload._scalar_definition,
-    datetime.date: base_scalars.Date._scalar_definition,
-    datetime.datetime: base_scalars.DateTime._scalar_definition,
-    datetime.time: base_scalars.Time._scalar_definition,
-    decimal.Decimal: base_scalars.Decimal._scalar_definition,
+    UUID: cast(_HasScalarDefinition, base_scalars.UUID)._scalar_definition,
+    Upload: cast(_HasScalarDefinition, Upload)._scalar_definition,
+    datetime.date: cast(_HasScalarDefinition, base_scalars.Date)._scalar_definition,
+    datetime.datetime: cast(
+        _HasScalarDefinition,
+        base_scalars.DateTime,
+    )._scalar_definition,
+    datetime.time: cast(_HasScalarDefinition, base_scalars.Time)._scalar_definition,
+    decimal.Decimal: cast(
+        _HasScalarDefinition,
+        base_scalars.Decimal,
+    )._scalar_definition,
```

Comment:
shall we make a a `_get_scalar_definition` function instead of doing all these casts and property access? :)

Reply:
Sure :)